src/arrayMethodSort.js

```diff
@@ -4,8 +4,38 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const compareFn =
+      compareFunction ||
+      function (a, b) {
+        const strA = String(a);
+        const strB = String(b);
+
+        if (strA < strB) {
+          return -1;
+        }
+
+        if (strA > strB) {
+          return 1;
+        }
+
+        return 0;
+      };
+
+    const len = this.length;
+
+    for (let i = 0; i < len - 1; i++) {
+      for (let j = 0; j < len - 1 - i; j++) {
```

Comment:
Consider optimizing the inner loop condition to avoid unnecessary comparisons. Since the largest elements 'bubble up' to the end of the array after each iteration of the outer loop, you can reduce the range of the inner loop by 'i'. This is already done, but you can further optimize it by using 'len - i' instead of 'len - 1 - i' because the last element is already in place and does not need to be compared.